### DISCUSSION TITLE
JSON:API / Drupal

### DISCUSSION THREAD

**Discussion by akuckartz** [SPARQL-Anything/sparql.anything#216] [Q&A]
Any idea on how to integrate data exposed using a [JSON:API](https://jsonapi.org/) based API ?

That would help to integrate data contained in Drupal-systems. See https://www.drupal.org/docs/core-modules-and-themes/core-modules/jsonapi-module

**Comment by enridaga** [comment]
For what I can see from the specification, APIS built using the JSON:API spec return a specific form of JSON. If the request can be made as an HTTP GET I guess it is sufficient to write a SERVICE clause such as:
```
x-sparql-anything:media-type=application/json,location=http:://domain-path-to-the-json-api-request
```
This will allow to query the content through Facade-X

**Comment by justin2004** [comment]
i was able to successfully do something like this today. the API endpoint returns json but i needed to add an HTTP header in the request (with an authorization token).

so i had to do a [fork](https://github.com/justin2004/sparql.anything/tree/http-header-support) which additionally treats all properties as HTTP headers.

then i just did something like:
```
SELECT *
WHERE {
    service <x-sparql-anything:media-type=application/json,Accept=application/json,x-auth-thing=secret_token_goes_here,https://some.com/api/users> {
        ?s ?p ?o }
}
```

**Comment by justin2004** [comment]
@akuckartz i think this works.

here is an example using the arxiv  REST API. it returns xml instead of json but sparql anything handles the json just fine too.

https://github.com/justin2004/weblog/tree/master/dynamic_pagination_with_sparql_anything

**Comment by enridaga** [comment]
I think we can mark this as solved.
